go/mysql/auth_server_static.go

```diff
@@ -52,10 +53,10 @@ type AuthServerStatic struct {
 	// It defaults to MysqlNativePassword.
 	Method string
 	// This mutex helps us prevent data races between the multiple updates of Entries.
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
There's no need for an RWMutex here -- it's not like the readers hold onto the lock for extended amounts of time, and the regular Mutex has better performance when not contended.